tests/integration-tests/remote_command_executor.py

```diff
@@ -27,15 +27,29 @@ def __init__(self, result):
 class RemoteCommandExecutor:
     """Execute remote commands on the cluster head node."""
 
-    def __init__(self, cluster, username=None):
+    def __init__(self, cluster, username=None, bastion=None):
         if not username:
             username = get_username_for_os(cluster.os)
-        self.__connection = Connection(
-            host=cluster.head_node_ip,
-            user=username,
-            forward_agent=False,
-            connect_kwargs={"key_filename": [cluster.ssh_key]},
-        )
+        connection_kwargs = {
+            "host": cluster.head_node_ip,
+            "user": username,
+            "forward_agent": False,
+            "connect_kwargs": {"key_filename": [cluster.ssh_key]},
+        }
+        if bastion:
+            # Need to execute simple ssh command before using Connection to avoid Paramiko _check_banner error
```

Comment:
Can you add a link if it's a known issue?

Reply:
Was not able to figure out the root cause of the issue. Most issue online about `Paramiko _check_banner` are about adjusting Paramiko `connect` parameters. However we use fabric connection in integration test, which depends on Paramiko, but do not support all Paramiko `connect` parameters to adjust `_check_banner` behaviors.

Didn't experiment with using Paramiko directly, since we do not want to change existing behavior of tests; I just came up with the work around.